### PR TITLE
A custom User model does not always have a username field.

### PR DESCRIPTION
**Pull request by tonioo**
See https://docs.djangoproject.com/en/1.8/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD

### CODE REVIEW

File: django_nyt/models.py [hunk 1]
```diff
@@ -116,7 +117,8 @@ class Subscription(models.Model):
     )
 
     def __str__(self):
-        obj_name = _("Subscription for: %s") % str(self.settings.user.username)
+        obj_name = _("Subscription for: %s") % str(
+            getattr(self.settings.user, self.settings.user.USERNAME_FIELD))
```
Comment (benjaoming) [inline django_nyt/models.py]: The `settings.user.USERNAME_FIELD` should be `settings.USERNAME_FIELD`, right?

**Comment by benjaoming** [comment]
Thanks for looking into this!

Would you be able to make the change so it's backwards compatible? Maybe in django_nyt.settings put:

    USERNAME_FIELD = getattr(settings, 'USERNAME_FIELD', 'username')

..and then use `django_nyt.settings` as a compatibility layer.

**Comment by tonioo** [comment]
Backward compatible with what? Django < 1.5 ?

**Comment by benjaoming** [comment]
Oh right, Django 1.4 is completely unsupported now, not even documentation is available anymore... nevermind that part then!

**Comment by tonioo** [comment]
Ok. Is there still something that prevents this PR to be merged ?

File: django_nyt/models.py (hunk 1 above)
Comment (benjaoming) [inline django_nyt/models.py]: Yes, this one still looks like an error ?

**Comment by benjaoming** [comment]
See commit comment...

https://github.com/benjaoming/django-nyt/pull/16/files#r35203092

File: django_nyt/models.py (hunk 1 above)
Comment (tonioo) [inline django_nyt/models.py]: Why? The USERNAME_FIELD attribute is on the User model...

Comment (benjaoming) [inline django_nyt/models.py]: Oh right, you see that's where I initially went wrong :)

**Comment by benjaoming** [comment]
Thanks for fixing this, sorry about slowness!

**Pull request merged**